contracts/modules/staking/PooledStaking.sol

```diff
@@ -737,47 +957,519 @@ contract PooledStaking is IPooledStaking, MasterAware {
+      return 15416480996133510000; // 15.41648099613351%
+    }
+
+    // Notional Finance v1
+    if (id == 55) {
+      return 33481175136260454000; // 33.481175136260454%
+    }
+
+    // Origin Dollar
+    if (id == 56) {
+      return 26859147414969534000; // 26.85914741496953%
+    }
+
+    // Opyn v2
+    if (id == 57) {
+      return 38049781406997820000; // 38.04978140699782%
+    }
+
+    // Vesper
+    if (id == 59) {
+      return 13546292639240680000; // 13.54629263924068%
+    }
+
+    // Benchmark Protocol
+    if (id == 60) {
+      return 18572683353728465000; // 18.57268335372847%
+    }
+
+    // Harvest Finance
+    if (id == 63) {
+      return 33869341091143590000; // 33.86934109114359%
+    }
+
+    // Alchemix v1
+    if (id == 67) {
+      return 20189390915194700000; // 20.1893909151947%
+    }
+
+    // Alpaca Finance
+    if (id == 70) {
+      return 9693452748544052000; // 9.69345274854405%
+    }
+
+    // Visor Finance
+    if (id == 71) {
+      return 26095937582591080000; // 26.095937582591077%
+    }
+
+    // Goldfinch
+    if (id == 72) {
+      return 4999785623702144000; // 4.999785623702144%
+    }
+
+    // Celsius
+    if (id == 73) {
+      return 15575757370822483000; // 15.575757370822483%
+    }
+
+    // inLock
+    if (id == 76) {
+      return 67405469242073440000; // 67.40546924207344%
+    }
+
+    // Pangolin
+    if (id == 86) {
+      return 16004943105245690000; // 16.00494310524569%
+    }
+
+    // Centrifuge Tinlake
+    if (id == 87) {
+      return 10622199086614927000; // 10.622199086614927%
+    }
+
+    // Premia Finance
+    if (id == 90) {
+      return 13609569933234702000; // 13.609569933234702%
+    }
+
+    // Bunny
+    if (id == 92) {
+      return 34957746687696930000; // 34.957746687696925%
+    }
+
+    // Venus
+    if (id == 93) {
+      return 39982787592238610000; // 39.982787592238616%
+    }
+
+    // Thorchain
+    if (id == 94) {
+      return 16340308447531193000; // 16.340308447531193%
+    }
+
+    // Pancakeswap v1
+    if (id == 95) {
+      return 29659851177897560000; // 29.659851177897558%
+    }
+
+    // Yearn yvDAI v2
+    if (id == 96) {
+      return 2681190691870258400; // 2.6811906918702584%
+    }
+
+    // Curve sETH LP (eCrv)
+    if (id == 100) {
+      return 3642751750084770400; // 3.6427517500847704%
+    }
+
+    // Idle DAI v4 (idleDAIYield)
+    if (id == 101) {
+      return 4831821144747536000; // 4.831821144747536%
+    }
+
+    // Idle USDT v4 (idleUSDTYield)
+    if (id == 102) {
+      return 5089820795846740000; // 5.08982079584674%
+    }
+
+    // Pool Together v4
+    if (id == 110) {
+      return 13592770586860942000; // 13.592770586860942%
+    }
+
+    // Trader Joe
+    if (id == 111) {
+      return 7995085516794736000; // 7.9950855167947354%
+    }
+    // {V1_PRICES_HELPER_END}
+
+    revert("Invalid product id");
+  }
+
+  function migrateToNewV2Pool(address stakerAddress, uint groupId) external noPendingActions {
```

Comment:
we should measure gas here to see if there's any issue with max block size and whether we need iterations.